File: BankKataJava/src/main/java/bankingkata/Account.java
package bankingkata;

public class Account {

    private final TransactionLog transactionLog;
    private StatementPrinter statementPrinter;

    public Account(TransactionLog transactionLog, StatementPrinter statementPrinter) {
        this.transactionLog = transactionLog;
        this.statementPrinter = statementPrinter;
    }

    public void withdraw(int amount) {
        transactionLog.addWithdraw(amount);
    }

    public void deposit(int amount) {
        transactionLog.addDeposit(amount);
    }

    public void printStatement() {
        statementPrinter.print(transactionLog.getAllTransactions());
    }

}

File: BankKataKotlin/src/main/kotlin/bankingkata/Account.kt
package bankingkata

class Account(
    private val transactionLog: TransactionLog,
    private val statementPrinter: StatementPrinter
) {

    fun withdraw(amount: Int) {
        transactionLog.addWithdraw(amount)
    }

    fun deposit(amount: Int) {
        transactionLog.addDeposit(amount)
    }

    fun printStatement() {
        statementPrinter.printAll(transactionLog.allTransactions)
    }
}

